table/block_based_table_builder.h

```diff
@@ -94,6 +93,15 @@ class BlockBasedTableBuilder : public TableBuilder {
  private:
   bool ok() const { return status().ok(); }
 
+  // True if currently buffering uncompressed data blocks in-memory. See
+  // `Rep::State` API comment for details of the states.
+  bool IsBuffered() const;
```

Comment:
I find the naming of this function a little less accurate. By its name, if `!IsBuffered`, the state can be either `kClosed` or `kUnbuffered`. In the program there are assertions on state not being `kClosed`, so when we say `!IsBuffered()`, we are really expecting `state == kUnbuffered`.

Reply:
Yeah that's true. Let me clean it up.